packages/react-native-audio-api/android/src/main/java/com/swmansion/audioapi/system/MediaNotificationManager.kt

```diff
@@ -182,45 +168,30 @@ class MediaNotificationManager(
     return NotificationCompat.Action(icon!!, title, i)
   }
 
-  class NotificationService : Service() {
-    private val binder = LocalBinder()
+  class AudioForegroundService : Service() {
     private var notification: Notification? = null
-
-    inner class LocalBinder : Binder() {
-      private var weakService: WeakReference<NotificationService>? = null
-
-      fun onBind(service: NotificationService) {
-        weakService = WeakReference(service)
-      }
-
-      fun getService(): NotificationService? = weakService?.get()
-    }
-
-    override fun onBind(intent: Intent): IBinder {
-      binder.onBind(this)
-      return binder
-    }
-
-    fun forceForeground() {
-      if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.O) {
-        val intent = Intent(this, NotificationService::class.java)
-        ContextCompat.startForegroundService(this, intent)
-        notification =
-          MediaSessionManager.mediaNotificationManager
-            .prepareNotification(NotificationCompat.Builder(this, MediaSessionManager.CHANNEL_ID), false)
-        startForeground(MediaSessionManager.NOTIFICATION_ID, notification)
-      }
-    }
-
-    override fun onCreate() {
-      super.onCreate()
-      try {
-        notification =
-          MediaSessionManager.mediaNotificationManager
-            .prepareNotification(NotificationCompat.Builder(this, MediaSessionManager.CHANNEL_ID), false)
-        startForeground(MediaSessionManager.NOTIFICATION_ID, notification)
-      } catch (ex: Exception) {
-        Log.w("AudioManagerModule", "Error starting service: ${ex.message}")
+    private var isServiceStarted = false
+    private val serviceLock = Any()
+
+    override fun onBind(intent: Intent): IBinder? = null
+
+    private fun startForegroundService() {
+      synchronized(serviceLock) {
+        if (!isServiceStarted) {
+          try {
+            notification =
+              MediaSessionManager.mediaNotificationManager
+                .prepareNotification(
+                  NotificationCompat.Builder(this, MediaSessionManager.CHANNEL_ID),
+                  false,
+                )
+            startForeground(MediaSessionManager.NOTIFICATION_ID, notification)
+            isServiceStarted = true
+          } catch (ex: Exception) {
+            Log.w("AudioManagerModule", "Error starting foreground service: ${ex.message}")
```

Comment:
if thats an error use `Log.e` instead
```suggestion
            Log.e("AudioManagerModule", "Error starting foreground service: ${ex.message}")
```